pkg/kv/mem/store.go

```diff
@@ -135,30 +170,31 @@ func (e *EntriesIterator) Next() bool {
 	if e.err != nil {
 		return false
 	}
-	if e.start == nil {
+	if e.start == "" {
 		e.entry = nil
 		return false
 	}
 	e.store.mu.RLock()
 	defer e.store.mu.RUnlock()
 	idx := sort.SearchStrings(e.store.keys, string(e.start))
 	if idx == len(e.store.keys) {
-		e.start = nil
+		e.start = ""
 		e.entry = nil
 		return false
 	}
 	// point to the entry we found
-	key := e.store.keys[idx]
-	value := e.store.m[key]
+	combinedKey := e.store.keys[idx]
+	value := e.store.m[combinedKey]
 	e.entry = &kv.Entry{
-		Key:   []byte(key),
-		Value: value,
+		PartitionKey: partitionKeyFromCombinedKey(combinedKey),
+		Key:          keyFromCombinedKey(combinedKey),
+		Value:        value,
 	}
 	// set start to the next item - nil as end indicator
 	if idx+1 < len(e.store.keys) {
-		e.start = []byte(e.store.keys[idx+1])
+		e.start = e.store.keys[idx+1]
```

Comment:
If the last key in a partition is reached, `Next()` continues to the next partition (right?). Is that the desired behavior? I believe the postgress implementation behaves differentley

Reply:
It's a test-only implementation (mem) and I'm assuming we won't have partitions with `_` in them. If we will have them, we can have a different separator (like `_*&^%$`).